### DISCUSSION TITLE
Configurable life time for RKE2 client and server certificates

### DISCUSSION THREAD

**Discussion by rajivml** [rancher/rke2#6639] [Q&A] [answered]
HI

This is the default behaviour of RKE2 certificates where RKE2 client and server certificates are valid for 365 days from their date of issuance. Any certificates that are expired, or within 90 days of expiring, are automatically renewed every time RKE2 starts.

But what we are looking for is an option through which we can configure the lifetime of RKE2 certificates because many of customers rarely upgrade and majority of the tickets we receive from customers are certificates related. So it would be great if there is an option exposed through which we can override the default lifetime of certificates

Now you guys may suggest us to generate our own certificates but managing these certificates is quite a lot of pain and we don't want to re-invent the wheel. All we are looking for is an option through which we can extend the default lifetime of the certificates

**Comment by rajivml** [comment]
@brandond is it something in works or on your team's radar ?

**Comment by rajivml** [comment]
We realised that there is a hacky way to achieve the same, but is it officially supported ? can the same be exposed via a flag keeping one year as default ? 
 
<img width="1237" alt="image" src="https://github.com/user-attachments/assets/fe8bcea8-e72d-4e34-81ec-8a0c967b29ff">

**Reply by brandond** [reply, marked as answer]
Use of `CATTLE_NEW_SIGNED_CERT_EXPIRATION_DAYS` is NOT officially supported, and we DO NOT document it or recommend it. However, we have no plans to remove support for this env var from the library that RKE2 uses to generate certificates.

**Comment by rajivml** [comment]
Thanks Brandon for confirmation that the support for this flag will not be removed in later releases

**Comment by xyzzyz** [comment]
@brandond so what's the recommended solution? The status quo is that rke2 breaks after running for 365 days without restart. Should we create cronjobs to restart rke2-server monthly/weekly to make sure the rotation is triggered?

**Reply by brandond** [reply]
Yes. You should be patching your nodes at least every few months. After a full year your Kubernetes minor version would be end of life and long overdue for an upgrade.

Kubernetes is not a good choice for those who want to install things and then never touch it again.

**Reply by maslo64** [reply]
There seems to be official way to adjust the value. 
This is now documented by SuSE here: https://support.scc.suse.com/s/kb/How-to-extend-RKE2-K3s-self-signed-certificate-expiration?language=en_US

**Reply by brandond** [reply]
Did you skip over reading the selected answer?

I will work with the support team to make sure that KB doc mentions that we do NOT support overriding the defaults via that env var; anyone changing it does so at their own risk.

**Reply by maslo64** [reply]
Hi Brad, reason why I placed the link is that I would consider this option now as supported and documented as it's now published by vendor, or I'm missing something ? 
Thanks & Best regards

**Reply by brandond** [reply]
No, it is not supported by the product team and intentionally is not covered in the RKE2 docs. KBs are authored by the support org and do not always get full engineering review. I am working with the support team to update the doc appropriately.
